guide/14-deep-learning/FCN_MLModel_guide-final_draft.ipynb

```diff
@@ -0,0 +1,517 @@
+{
```

Comment:
<p>Prepare tabular data</p><p> </p><p>Data preparation can be a time consuming process that typically involves splitting the data into training and validation sets, applying various data augmentation techniques, creating the necessary data structures for loading data into the model, memory management by using the appropriately sized mini-batches of data and so on.</p><p> </p><p>You can use <code>prepare_tabulardata</code>&nbsp;method to create a TabularDataObject that can be fed into the MLModel and the fully connected network models. It allows you to use attributes data from feature layers and spatial dataframes. Additionally, you can use explanatory variables from rasters. The rasters can be local rasters through the raster class or imagery layer. This method performs normalization of the datasets, imputation and splits it into training and validation datasets before feeding into the model.</p><p>The <code>prepare_tabulardata</code> method introspects the type of fields in the input data and automatically treats them as continuous or categorical variables. However, it also gives flexibility to overwrite this default behavior using the following example:</p><p> </p><p>For example if <em>n</em> predictors are used and one is categorical and the other continuous then:</p><pre class="ql-syntax" spellcheck="false">X = ["Field_1", ("Field_2", True),......."Field_n"] 
</pre><p>Here Field_2 is treated as categorical and the others as continuous variables</p><p> </p><p>In case, these variables are stored as multiple rasters then,</p><pre class="ql-syntax" spellcheck="false">explanatory_rasters = [RasterClassObject1,(RasterClassObject2, True),.......RasterClassObjectn] 
</pre><p>Here RasterClassObject2 is treated as categorical and the others as continuous variables</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/718/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>